src/main/java/co/rsk/federate/FederatorSupport.java

```diff
@@ -151,6 +161,80 @@ public StateForFederator getStateForFederator() {
         return new StateForFederator(result, this.parameters);
     }
 
+    public Map<String, Utxo> getUtxosAtBlock(long atBlockNumber) {
+
+        String utxosJson = getUtxoJsonAtBlockRcpCall(atBlockNumber);
+
+        List<Utxo> utxos = Utxo.parseRLPToActiveFederationUtxos(utxosJson);
+
+        Map<String, Utxo> utxoMap = utxos.stream().collect(Collectors.toMap(Utxo::getBtcTxHash, utxo -> utxo));
+
+        return utxoMap;
+
+    }
+
+    private String getUtxoJsonAtBlockRcpCall(long atBlockNumber) {
+        try {
+
+            URL url = new URL("http://localhost:4450");
```

Comment:
As seen here, this example only works for a local node running on port 4450. This is an example to show the POC, no need to do an actual implementation at the moment.

Reply:
Also, we could find a better way to do the rpc call.